### PR TITLE
Introduce new update test build agent

### LINKED ISSUE #5873: Split update-tests from integration-tests to a different Jenkins agent in the CI pipeline
**Description**

Both the update tests and the integration tests modules make heavy use of resources and ad hoc containers. To avoid port collisions and resource usage contention, we should split them up as we did for the integration tests from the rest of the pipeline in this PR https://github.com/zeebe-io/zeebe/pull/5735.

The assumption is that this will reduce the flakiness of our tests as well.

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR introduces a new update test build agent. It also defines a new podSpec for each agent and attempts to balance resources properly. I think we still overprovision, but I erred on the side of caution here just in case.

1. The `distribution` pod spec now still has a dind daemon, but much smaller as we don't expect to make such heavy use of it. Modules making use of it are, for example, the Elasticsearch exporter, the Gateway, etc., all for the integration tests. It might be a good idea in the future to also run integration tests from other modules as part of the integration stage anyway, but I would say that's outside of scope here.
1. The `update-test` pod spec has only the maven and dind containers, and most of the resources are for the dind container.
1. The `integration-test` pod spec is the same but with a different split of resources, since the maven container there needs more.

I tried initially to parallelize all the steps, but as agents are not reused across stages (and `idleMinutes` was a no go since the pods are not tied to the lifecycle of the job).

As a consequence, I ended up switching so that a single verification stage which is parallel, and inside we have the usual stages (e.g. BPMN TCK, unit tests, etc.), but the integration and update ones are split to a different agents, and comprised of multiple stages.

One thing which might seem odd - we build the docker image before the parallel stage. That's because the various parallel stages on the main pod require the docker image to be built. Unfortunately, we can't nest parallel stages inside a parallel stage, so to run the IT and update tests at the same time as we build the original docker image, we'd need to make all the other stages sequential, which isn't ideal.

I also added DOCKER_BUILDKIT to the environment when building the docker images to speed it up, so having it copied is not much overhead (~30s).

## Related issues

closes #5873

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by npepinpe** [comment]
@cmur2 if you're busy no worries :slightly_smiling_face: I just added you so maybe you can give it a sanity check, primarily in the Jenkinsfile, maybe there's something wrong I'm doing

File: Jenkinsfile [hunk 1]
```diff
@@ -22,7 +22,10 @@ pipeline {
             cloud 'zeebe-ci'
             label "zeebe-ci-build_${buildName}"
             defaultContainer 'jnlp'
-            yamlFile '.ci/podSpecs/distribution.yml'
+            yamlFile ".ci/podSpecs/distribution.yml"
+            // allows the pod to remain active for reuse until the configured number of minutes
+            // has passed since the last step was executed on it.
+            idleMinutes 15
```
Comment (cmur2) [inline Jenkinsfile:28]: I think this should be removed based on the PR's description - it's not needed and will leave the pod running for 15min after a build finished but due to using the `label "zeebe-ci-build_${buildName}"` only one build will ever use this pod.

**Review by cmur2 — APPROVED** [review]
LGTM, I think the pipeline looks very modular now with all stages mostly doing some `unstash` + one command consuming this.

**Comment by cmur2** [comment]
> As for most resource limits/requests, I based it off what I saw the pods using (so if one was maxxing out its CPU, I would increase it so it would always have a little bit of buffer, and same with memory).

@npepinpe I saw that in general you reduced memory usage limits more often than CPU usage limits.
This is great but don't force yourself too much saving on the memory usage limit as memory is [really cheap](https://cloud.google.com/compute/vm-instance-pricing) compared to CPU cores, at a rate of roughly 1:10 given that Google says: $0.036489 / vCPU hour and $0.004892 / GB hour.
Also the [machine type](https://cloud.google.com/compute/docs/machine-types) your builds are requesting (n1-standard-32) provides in total 32 vCPU and 120 GiB memory so a default quota ratio would be 1:3.75 (vCPU to memory) -- not saying that you have to do this, just so you don't think too hard about the memory limit as the memory is payed for anyways using fixed machine types.

(In the longterm Infra team might decide to switch the machine type if we observe that (would be typical for a CI) the ratio between vCPU and memory stays at more like 1:2 instead of 1:3.75 for n1-standard type.)

> I had some trouble with the IT stage where if I gave too little memory the forked VM kept crashing? Never figured out why tbh, but increasing it fixed it =/ I'd be happy if someone knew and could point out the reason/proper fix.

I guess to properly fix it we'd have to delve into the details of how JVM works in containerized environments and how to configure that for the respective JVM version (8 has different features than 11), e.g. via https://stackoverflow.com/questions/49854237/is-xxmaxramfraction-1-safe-for-production-in-a-containered-environment

I'd recommend we do that at some point, maybe track it via an issue?

File: Jenkinsfile [hunk 2]
```diff
@@ -142,48 +156,104 @@ pipeline {
                     }
                 }
 
-                stage('IT (Java)') {
+                stage('Integration') {
```
Comment (korthout) [inline Jenkinsfile:159]: Can we rename this to `IT` and `Update Test` to `Update`? Currently, it get's truncated on my relatively large laptop screen (16" macbook pro).
Screenshot 2020-11-25 at 10 03 35

**Review by korthout — APPROVED** [review]
This is a really nice change. The Jenkinsfile changes are clean and it does what it should. 

I have to some nitpicks, but here's my approval 🎉 

PS: 

> I also added DOCKER_BUILDKIT to the environment when building the docker images to speed it up, so having it copied is not much overhead (~30s).

I love it.

EDIT: Shouldn't this be backported?

File: .ci/podSpecs/distribution.yml [hunk 3]
```diff
@@ -8,9 +8,6 @@ spec:
     - key: "agents-n1-standard-32-netssd-preempt"
       operator: "Exists"
       effect: "NoSchedule"
-  volumes:
-    - name: shared-data
-      emptyDir: {}
   containers:
     - name: maven
       image: maven:3.6.0-jdk-11
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply .ci/podSpecs/distribution.yml:13]: That's a good point, we should :+1:

File: Jenkinsfile (hunk 2 above)
Comment (npepinpe) [reply Jenkinsfile:159]: truncation? not in my pipeline! :sunglasses: 

but yeah sure :smile:

File: Jenkinsfile (hunk 1 above)
Comment (npepinpe) [reply Jenkinsfile:28]: Ah, I think this is a leftover from rebasing very often :see_no_evil:

File: .ci/podSpecs/distribution.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (korthout) [reply .ci/podSpecs/distribution.yml:13]: In that case, note that this image is used in multiple specs.

File: Jenkinsfile (hunk 2 above)
Comment (korthout) [reply Jenkinsfile:159]: Do you use a 48" OLED TV as monitor? 😆

Comment (Zelldon) [reply Jenkinsfile:159]: UltraWide :cool:

Comment (npepinpe) [reply Jenkinsfile:159]: I code in the only acceptable way, in IMAX of course

**Comment by npepinpe** [comment]
@cmur2 regarding our resources, I used k9s and saw that we were mostly CPU bound, and we often had a lot of memory unused. To be honest, I think the best would be to gather historical metrics so we could adjust these over time, as I'm not 100% sure of the requests at the moment, they're very much estimates.

**Comment by npepinpe** [comment]
Eh, looked at that, updating Go image creates non backwards compatible changes :sweat_smile:

**Review by korthout — APPROVED** [review]
LGTM 🌮

**Comment by npepinpe** [comment]
> The process '/home/runner/work/_actions/zeebe-io/backport-action/master/backport.sh' failed with exit code 4

> The process '/home/runner/work/_actions/zeebe-io/backport-action/master/backport.sh' failed with exit code 4

[noooooo](http://www.nooooooooooooooo.com/)

**Comment by korthout** [comment]
> exit code 4

These are git conflicts

**Pull request merged**